packages/frontend/src/components/wallet-migration/WalletMigration.jsx

```diff
@@ -57,25 +69,33 @@ const WalletMigration = ({ open, onClose }) => {
         handleStateUpdate({ walletType });
     };
 
-    const handleSetActiveView = (activeView) => {
+    const handleSetActiveView = useCallback((activeView) => {
         handleStateUpdate({ activeView });
-    };
+    }, [handleStateUpdate]);
 
     const showMigrationPrompt = useCallback(() => {
         handleSetActiveView(WALLET_MIGRATION_VIEWS.MIGRATION_PROMPT);
-    }, []);
+    }, [handleSetActiveView]);
 
-    const showMigrateAccount = useCallback(async () => {
+    const showMigrateAccount = useCallback(() => {
         handleSetActiveView(WALLET_MIGRATION_VIEWS.MIGRATE_ACCOUNTS);
-    }, [availableAccounts]);
+    }, [handleSetActiveView]);
 
     const onContinue = useCallback(async () => {
-        const url = await encodeAccountsToURL(
-            availableAccounts,
-            state.migrationKey
-        );
+        let url = '';
+        if (state.walletType === 'sender') {
+            url = await encodeAccountsToSender(
```

Comment:
why duplicate the same function?

Reply:
Shouldn't block this PR based on this. Will address in a separate PR.

Reply:
None of this will have to change if we do the minor refactor described in https://github.com/near/near-wallet/pull/2883/files#r946033135